services/terrain/src/terrain/services/filesystem/metadata.clj

```diff
@@ -377,44 +200,59 @@
 (defn- bulk-add-avus
   "Applies metadata from a list of attributes and filename/values to those files found under
    dest-dir."
-  [cm user dest-dir force? template-id template-attrs attrs csv-filename-values]
+  [user dest-dir force? template-id template-attrs attrs csv-filename-values]
   (let [format-path (partial format-csv-metadata-filename dest-dir)
         paths (map (comp format-path first) csv-filename-values)
+        path-info-map (-> (data-raw/collect-stats user :paths paths :validation-behavior "write") :body json/decode (get "paths"))
         value-lists (map rest csv-filename-values)
         irods-attrs (clojure.set/difference (set attrs) (set (keys template-attrs)))]
-    (validators/all-paths-exist cm paths)
-    (validators/all-paths-writeable cm user paths)
     (if-not force?
-      (validate-batch-add-attrs cm paths irods-attrs))
-  (mapv (partial bulk-add-file-avus cm user dest-dir template-id template-attrs attrs)
-    paths value-lists)))
+      (validate-batch-add-attrs user (map #(get-in path-info-map [% "id"]) paths) irods-attrs))
+  (mapv (partial bulk-add-file-avus user dest-dir template-id template-attrs attrs)
+    path-info-map value-lists)))
+
+(defn- keyword-to-int [kw] (Integer/parseInt (name kw)))
+
+(defn- deparse-csv-line
+  [line]
+  (mapv second (into (sorted-map-by #(< (keyword-to-int %1) (keyword-to-int %2))) line)))
```

Comment:
This function basically inverts the code in data-info which formats the CSV into something nice for the UI, and turns it back into what opencsv would have returned originally. That is, from a vector of maps from column numbers to column contents, it turns them back into ordered vectors of column contents without explicit column numbers.